plugins/modules/ec2_vpc_igw.py

```diff
@@ -217,7 +217,8 @@ def ensure_igw_present(self, vpc_id, tags, purge_tags):
             self._connection, self._module, igw['internet_gateway_id'],
             resource_type='internet-gateway', tags=tags, purge_tags=purge_tags
         )
-        igw_info = self.get_igw_info(self.get_matching_igw(vpc_id), vpc_id)
+        igw = self.get_matching_igw(vpc_id)
```

Comment:
This fixes “TypeError: 'NoneType' object is not subscriptable”.

Reply:
```suggestion

```
@alinabuzachis pretty sure this line can be removed as `igw` is defined earlier on in the method.. worked for me at least!